### ISSUE TITLE
Resources tooltips - Missing items

### ISSUE THREAD

**Issue by GreatApo** [DinoDevs/GladiatusCrazyAddon#263]
![image](https://user-images.githubusercontent.com/9620510/115093292-bb442d80-9f11-11eb-85f5-4ba852fdc012.png)

**Comment by GreatApo** [comment]
The bug only happens when resources style is minimal-amounts' or 'extended-amounts'. [This callback](https://github.com/DinoDevs/GladiatusCrazyAddon/blob/master/source/core/source/global.js#L3482) never replies. I don't know how to debug the background page, so @GramThanos have fun.

**Comment by PatrykLeczycki** [comment]
> The bug only happens when resources style is 'minimal-amounts' or 'extended-amounts'. [This callback](https://github.com/DinoDevs/GladiatusCrazyAddon/blob/master/source/core/source/global.js#L3482) never replies. I don't know how to debug the background page, so @GramThanos have fun.

I'm not sure what are these resources styles you referred to, but as I see the bug happens when given item doesn't have sufix. On my level I can't find items without prefixes on auctions, but maybe someone could confirm if the same problem happens when an item doesn't have prefix but it has only sufix.

**Comment by GreatApo** [comment]
> > The bug only happens when resources style is 'minimal-amounts' or 'extended-amounts'. [This callback](https://github.com/DinoDevs/GladiatusCrazyAddon/blob/master/source/core/source/global.js#L3482) never replies. I don't know how to debug the background page, so @GramThanos have fun.
> 
> I'm not sure what are these resources styles you referred to, but as I see the bug happens when given item doesn't have sufix. On my level I can't find items without prefixes on auctions, but maybe someone could confirm if the same problem happens when an item doesn't have prefix but it has only sufix.

Not exactly. This item bellow doesn't have a suffix:
![image](https://user-images.githubusercontent.com/9620510/115708114-7f70f400-a367-11eb-964a-3dba43b77757.png)

However, you may visually not see a suffix, but there are suffixes without names... That may be related to the problem.

PS. Resources styles are the different ways you can display the resources at the tooltips.

**Comment by PatrykLeczycki** [comment]
> > > The bug only happens when resources style is 'minimal-amounts' or 'extended-amounts'. [This callback](https://github.com/DinoDevs/GladiatusCrazyAddon/blob/master/source/core/source/global.js#L3482) never replies. I don't know how to debug the background page, so @GramThanos have fun.
> > 
> > 
> > I'm not sure what are these resources styles you referred to, but as I see the bug happens when given item doesn't have sufix. On my level I can't find items without prefixes on auctions, but maybe someone could confirm if the same problem happens when an item doesn't have prefix but it has only sufix.
> 
> Not exactly. This item bellow doesn't have a suffix:
> ![image](https://user-images.githubusercontent.com/9620510/115708114-7f70f400-a367-11eb-964a-3dba43b77757.png)
> 
> However, you may visually not see a suffix, but there are suffixes without names... That may be related to the problem.
> 
> PS. Resources styles are the different ways you can display the resources at the tooltips.

You're right, I found some items without visible sufixes and tried to analyze their levels. It seems that when item's level suggest that it has any of +1 to +10 sufix, resources are visible, but the're not visible when level indicates that there's definitely no sufix. As you wrote at the beginning, some display options partially solve this problem, but they don't show amount of resources, so it's not a good solution in my case.

**Comment by GreatApo** [comment]
> As you wrote at the beginning, some display options partially solve this problem, but they don't show amount of resources, so it's not a good solution in my case.

I didn't said it is a solution, I traced where the issue is in the code. I just couldn't investigate further because the error is coming from a script running in browser's background (because the resources data are just to big in size to load them with each refresh) which I don't know how to debug (DarkThanos will investigate it). I pointed out that the other styles work to highlight that the resources count code it probably the problem. Without even looking the code, I suspect that the suffix is not contributing to the resources and in the data list is either commented or has a value of -1, which could lead to an error thrown in the background page and not in the Gladiatus webpage console ;)

I hope its clear now :)

**Comment by PatrykLeczycki** [comment]
> > As you wrote at the beginning, some display options partially solve this problem, but they don't show amount of resources, so it's not a good solution in my case.
> 
> I didn't said it is a solution, I traced where the issue is in the code. I just couldn't investigate further because the error is coming from a script running in browser's background (because the resources data are just to big in size to load them with each refresh) which I don't know how to debug (DarkThanos will investigate it). I pointed out that the other styles work to highlight that the resources count code it probably the problem. Without even looking the code, I suspect that the suffix is not contributing to the resources and in the data list is either commented or has a value of -1, which could lead to an error thrown in the background page and not in the Gladiatus webpage console ;)
> 
> I hope its clear now :)

Of course, thanks for explanation  :)
